src/components/PostDetails.tsx

```diff
@@ -65,28 +49,18 @@ export const PostDetails: React.FC<Props> = ({ post }) => {
         postId: post.id,
       });
 
-      setComments(
-        currentComments => [...currentComments, newComment],
-      );
-
-      // setComments([...comments, newComment]);
-      // works wrong if we wrap `addComment` with `useCallback`
-      // because it takes the `comments` cached during the first render
-      // not the actual ones
+      dispatch(setComments([...comments, newComment]));
     } catch (error) {
-      // we show an error message in case of any error
-      setError(true);
+      dispatch(setHasError(true));
     }
   };
 
   const deleteComment = async (commentId: number) => {
-    // we delete the comment immediately so as
-    // not to make the user wait long for the actual deletion
-    setComments(
-      currentComments => currentComments.filter(
+    dispatch(setComments(
+      comments.filter(
```

Comment:
Similar to the previous comment, instead of directly modifying the state by passing the new array to `setComments`, it's better to create an action in the commentsSlice to remove a comment from the state.